dashboard/modules/tune/tune_head.py

```diff
@@ -0,0 +1,233 @@
+                with open(error_path) as f:
+                    text = f.read()
+                    self._errors[str(trial)] = {
+                        "text": text,
+                        "job_id": os.path.basename(self._logdir),
+                        "trial_id": "No Trial ID"
+                    }
+                    other_data = df[df["logdir"].str.contains(trial)]
+                    if len(other_data) > 0:
+                        trial_id = other_data["trial_id"].values[0]
+                        self._errors[str(trial)]["trial_id"] = str(trial_id)
+                        if str(trial_id) in self._trial_records.keys():
+                            self._trial_records[str(trial_id)]["error"] = text
+                            self._trial_records[str(trial_id)][
+                                "status"] = "ERROR"
+
+    @async_loop_forever(tune_consts.TUNE_STATS_UPDATE_INTERVAL_SECONDS)
+    async def collect(self):
+        """
+        Collects and cleans data on the running Tune experiment from the
+        Tune logs so that users can see this information in the front-end
+        client
+        """
+        self._trial_records = {}
+        self._errors = {}
+        if not self._logdir:
+            return
+
+        # search through all the sub_directories in log directory
+        analysis = Analysis(str(self._logdir))
+        df = analysis.dataframe(metric="episode_reward_mean", mode="max")
+
+        if len(df) == 0 or "trial_id" not in df.columns:
+            return
+
+        self._trials_available = True
+
+        # make sure that data will convert to JSON without error
+        df["trial_id_key"] = df["trial_id"].astype(str)
+        df = df.fillna(0)
+
+        trial_ids = df["trial_id"]
+        for i, value in df["trial_id"].iteritems():
+            if type(value) != str and type(value) != int:
+                trial_ids[i] = int(value)
+
+        df["trial_id"] = trial_ids
+
+        # convert df to python dict
+        df = df.set_index("trial_id_key")
+        trial_data = df.to_dict(orient="index")
+
+        # clean data and update class attribute
+        if len(trial_data) > 0:
+            trial_data = self.clean_trials(trial_data)
+            self._trial_records.update(trial_data)
+
+        self.collect_errors(df)
+
+    def clean_trials(self, trial_details):
+        first_trial = trial_details[list(trial_details.keys())[0]]
+        config_keys = []
+        float_keys = []
+        metric_keys = []
+
+        # list of static attributes for trial
+        default_names = [
+            "logdir", "time_this_iter_s", "done", "episodes_total",
+            "training_iteration", "timestamp", "timesteps_total",
+            "experiment_id", "date", "timestamp", "time_total_s", "pid",
+            "hostname", "node_ip", "time_since_restore",
+            "timesteps_since_restore", "iterations_since_restore",
+            "experiment_tag", "trial_id"
+        ]
```

Comment:
If we don't care the order of the names, can we change the `default_names` to a set?
```
timeit.timeit("'iterations_since_restore' in default_names_set", globals=globals())
0.04095575900282711
timeit.timeit("'iterations_since_restore' in default_names", globals=globals())
0.2835074480099138
```